js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,183 @@
+
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+*
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+*                            Если длина 10 -> 'you win'
+*
+*
+*
+*
+* */
+
+function returnValueFromTypeOfData (argum) {
+    
+    if(typeof argum === 'number' || typeof argum === 'object') {
+        return true;
+    }
+    
+    if(typeof argum === 'function') {
+        return false;
+    }
+    
+    if(argum.length !== 10 && typeof argum === 'string') {
+        return `Длина вашей строки: ${argum.length}`;
+    }
+    
+    if(argum.length === 10 && typeof argum === 'string') {
+        return `you win`;
+    }
+}
+
+console.log(returnValueFromTypeOfData(1));
+console.log(returnValueFromTypeOfData({}));
+console.log(returnValueFromTypeOfData(function someFunction () {}));
+console.log(returnValueFromTypeOfData(`12345678`));
+console.log(returnValueFromTypeOfData(`123456789`));
+console.log(returnValueFromTypeOfData(`1234567890`));
+
+
+/*
+*
+*  Задача в классе !
+*
+* */
+const webStore = {
+    catagory: {
+        noteBook: [`ASUS`, `Lenovo`, `Dell`, `HP`],
+        mobile: [`Nokia`, `Apple`, `Sony`],
+    }
+}
+
+const category = webStore.catagory;
+
+function catagoryFromWebStore (obj) {
+    
+    for(let key in category) {
+        
+        console.log(`Категория:`, key);
+        
+        for(let i = 0; i < category[key].length; i++) {
+            
+            console.log(`Содержимое категории:`, category[key][i]);
+            
+        }
+    }
+}
+
+catagoryFromWebStore(category);
+
+/*
+ 1. Напишите функцию которая принимает 2 числа
+ и возвращает массив содержащий числа между первым числом и вторым числом;
+ */
+
+function numbersBetween(a, b) {
+    
+    for(let i = a; i <= b; i++) {
+       console.log(i);
+    }
+    
+}
+
+numbersBetween(3, 5);
+// 3, 4, 5
+
+numbersBetween(10, 12);
+// 10, 11, 12
+
+numbersBetween(20, 25);
+
+
+
+/*
+ 2. Перепишите задачу FizzBuzz, но с использованием цикла.
+ Расчет чисел должен идти до 100
+ */
+
+// 1. FizzBuzz 3, 5, 3 && % 5
+
+function FizzBuzz(number) {
+    
+    if(number % 3 === 0 && number % 5 === 0) {
```

Comment:
we reviewed solution on FizzBuzz in classwork, I guess you may apply such approach here